### PR TITLE
(546) - RoleDefinitions: Updated Module To Align with Other Resources

### PR DESCRIPTION
**Pull request by ahmadabdalla**
# Change

- Aligned to the [ARM Documentation](https://docs.microsoft.com/en-us/azure/templates/microsoft.authorization/roledefinitions?tabs=bicep)
- Introduced a new Parameter (assignableScopes) and added logic to process it.
- Used `@sys.description` instead of `description` so that we can access the same parameter name in the documentation. So `roleDescription` is now `description`.
- Added description to the outputs.
- Removed `dataActions` and `notDataActions` from the management group scope deployment as they are not supported. 
- Added a `min.parameters.json` so that it captures the minimum content for a role assignment.
- Expanded the `parameters.json` to cater for all parameter files in the template.
- Output changed from `roleDefinitionId` to `roleDefinitionResourceId`
- Updated Readme documentation to cater for new changes.
- Aligned to other modules for module names within the deploy.bicep
- Removed the location from the GUID name generated for Role Definitions as they are not pinned to locations.

## Testing

[Authorization: Roledefinitions](https://github.com/Azure/ResourceModules/actions/workflows/ms.authorization.roledefinitions.yml)

- Testing has been performed on the nested modules locally. Did an nested_mg, nested_sub, nested_rg.
- Testing has been performed on the main deploy.bicep targeting mg, sub and rg

## Type of Change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: arm/Microsoft.Authorization/roleDefinitions/.bicep/nested_roleDefinitions_mg.bicep [hunk 1]
```diff
@@ -1,34 +1,28 @@
 targetScope = 'managementGroup'
 
 param roleName string
-param roleDescription string = ''
+param description string = ''
```
Comment (eriqua) [inline arm/Microsoft.Authorization/roleDefinitions/.bicep/nested_roleDefinitions_mg.bicep:4]: I'd suggest adding @description for param and output also for nested modules

File: arm/Microsoft.Authorization/roleDefinitions/deploy.bicep [hunk 2]
```diff
@@ -1,81 +1,85 @@
...
     resourceGroupName: resourceGroupName
-    location: location
+    assignableScopes: assignableScopes
   }
 }
 
+@sys.description('The GUID of the Role Definition')
 output roleDefinitionName string = !empty(managementGroupId) ? roleDefinition_mg.outputs.roleDefinitionName : (!empty(resourceGroupName) ? roleDefinition_rg.outputs.roleDefinitionName : roleDefinition_sub.outputs.roleDefinitionName)
-output roleDefinitionId string = !empty(managementGroupId) ? roleDefinition_mg.outputs.roleDefinitionId : (!empty(resourceGroupName) ? roleDefinition_rg.outputs.roleDefinitionId : roleDefinition_sub.outputs.roleDefinitionId)
+@sys.description('The Resource ID of the Role Definition')
```
Comment (eriqua) [inline arm/Microsoft.Authorization/roleDefinitions/deploy.bicep:82]: I'd suggest to add new lines between outputs and param

File: arm/Microsoft.Authorization/roleDefinitions/deploy.bicep [hunk 3]
```diff
@@ -1,81 +1,85 @@
 targetScope = 'managementGroup'
 
-@description('Required. Name of the custom RBAC role to be created.')
+@sys.description('Required. Name of the custom RBAC role to be created.')
```
Comment (eriqua) [inline arm/Microsoft.Authorization/roleDefinitions/deploy.bicep:3]: Please change `roleName` to `name` here and in all other relevant files

Comment (ahmadabdalla) [reply arm/Microsoft.Authorization/roleDefinitions/deploy.bicep:3]: This is aligning to the doc https://docs.microsoft.com/en-us/azure/templates/microsoft.authorization/roledefinitions?tabs=bicep

File: arm/Microsoft.Authorization/roleDefinitions/.bicep/nested_roleDefinitions_mg.bicep (hunk 1 above)
Comment (ahmadabdalla) [reply arm/Microsoft.Authorization/roleDefinitions/.bicep/nested_roleDefinitions_mg.bicep:7]: Fixed

File: arm/Microsoft.Authorization/roleDefinitions/deploy.bicep (hunk 2 above)
Comment (ahmadabdalla) [reply arm/Microsoft.Authorization/roleDefinitions/deploy.bicep:82]: added

File: arm/Microsoft.Authorization/roleDefinitions/deploy.bicep (hunk 3 above)
Comment (MrMCake) [reply arm/Microsoft.Authorization/roleDefinitions/deploy.bicep:3]: As our motivation was to stick to the API to use name, I'd suggest to keep `roleName` here

Comment (ahmadabdalla) [reply arm/Microsoft.Authorization/roleDefinitions/deploy.bicep:3]: @eriqua , thoughts on the above please?

Comment (eriqua) [reply arm/Microsoft.Authorization/roleDefinitions/deploy.bicep:3]: Of course fine in that case

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**